rest-server/src/controllers/job.js

```diff
@@ -26,77 +27,42 @@ const load = (req, res, next, jobName) => {
   new Job(jobName, (job, error) => {
     if (error) {
       if (error.message === 'JobNotFound') {
-        if (req.method !== 'PUT') {
-          logger.warn('load job %s error, could not find job', jobName);
-          return res.status(404).json({
-            error: 'JobNotFound',
-            message: `could not find job ${jobName}`,
-          });
+        if (req.method !== 'PUT' && req.method !== 'POST') {
+          error.status = httpStatus.NOT_FOUND;
+          error.message = `could not find job ${jobName}`;
+          next(error);
         }
       } else {
-        logger.warn('internal server error');
-        return res.status(500).json({
-          error: 'InternalServerError',
-          message: 'internal server error',
-        });
+        error.status = httpStatus.INTERNAL_SERVER_ERROR;
```

Comment:
>error [](start = 8, length = 5)

in this case, internal server error is not helpful. can you include the error stack in error.message?

Reply:
This status equals to setting `500` http code, the original error will be returned directly. If `NODE_ENV` env var is set to `development`, the stack will be included.

Reply:
please change the default env to development then.
here: [https://github.com/Microsoft/pai/blob/8cff97104b6becd9dd761afaceef6e6df4632c20/pai-management/src/rest-server/dockerfile.template#L32](https://github.com/Microsoft/pai/blob/8cff97104b6becd9dd761afaceef6e6df4632c20/pai-management/src/rest-server/dockerfile.template#L32)


---
In reply to: [193369399](https://github.com/Microsoft/pai/pull/660#discussion_r193369399) [](ancestors = 193369399)

Reply:
Dumping error messages to log is not sufficient. Front-end applications can’t access the log, so we need to output something more meaningful directly in API response body.

Reply:
“returned directly” means it will be returned to http response